### PR TITLE
Youtube service

### PR DESCRIPTION
**Pull request by WyriHaximus**
Youtube service for #72

### CODE REVIEW

**Comment by WyriHaximus** [comment]
Ok while trying to get this sorted Google is frustrating me by refusing to accept my API key. Not sure what is going on but research in hacking together a small client myself to try and find the issue. This is annoying me for over a month.

File: src/AmsterdamPHP/Bundle/SiteBundle/Resources/config/services.yml [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
 services:
+    amsterdamphp_site.integration.youtube_factory:
+        class: %amsterdamphp_site.integration.youtube_factory.class%
+        arguments:
+            key: %youtube_key%
+
+    amsterdamphp_site.integration.youtube_client:
+        class:            "%amsterdamphp_site.integration.youtube_client.class%"
+        factory_service:  amsterdamphp_site.integration.youtube_factory
+        factory_method:   getClient
```
Comment (rdohms) [inline src/AmsterdamPHP/Bundle/SiteBundle/Resources/config/services.yml]: Ok i'm fine with this, even if you could drop the factory and just intantiate the client directly as a service, but this gives us a place to set defaults if we ever need them.

File: composer.json [hunk 2]
```diff
@@ -30,7 +30,8 @@
         "guzzle/guzzle": "~3.7",
         "ornicar/akismet-bundle": "*@dev",
         "ornicar/gravatar-bundle": "dev-master@dev",
-        "antimattr/google-bundle": "dev-master@dev"
+        "antimattr/google-bundle": "dev-master@dev",
```
Comment (pascaldevink) [inline composer.json]: Apparantly, animattr tagged a couple of versions in the mean time :grinning: https://packagist.org/packages/antimattr/google-bundle, can we use a stable version?

**Comment by pascaldevink** [comment]
Also, how on earth did you get a YouTube API key? :confounded: And what playlist value did you use?

**Comment by WyriHaximus** [comment]
You get the API key from: https://console.developers.google.com
The playlist value is: `UUAzioonGTZRKIP7qB_-P15w`

File: composer.json (hunk 2 above)
Comment (WyriHaximus) [inline composer.json]: Ow neat, I'll look into that :+1:

File: src/AmsterdamPHP/Bundle/SiteBundle/Resources/config/services.yml (hunk 1 above)
Comment (WyriHaximus) [inline src/AmsterdamPHP/Bundle/SiteBundle/Resources/config/services.yml]: After talking with you this was my assumption as how you preferred to have it setup. (Heck I took you meetup bundle as example.) If this is fine I'll let it be but if you really prefer instantiating the client directly I'll change it to that.

Comment (pascaldevink) [inline src/AmsterdamPHP/Bundle/SiteBundle/Resources/config/services.yml]: I think it's fine like this :+1:

Comment (rdohms) [inline src/AmsterdamPHP/Bundle/SiteBundle/Resources/config/services.yml]: I/m fine with this way.

File: composer.json (hunk 2 above)
Comment (WyriHaximus) [inline composer.json]: After looking at the package name it's a different package then what is used for youtube. Maybe it's out of scope for this PR?

File: src/AmsterdamPHP/Bundle/SiteBundle/Resources/config/services.yml (hunk 1 above)
Comment (WyriHaximus) [inline src/AmsterdamPHP/Bundle/SiteBundle/Resources/config/services.yml]: Great! :+1:

**Comment by pascaldevink** [comment]
Checked it out and it is a great service! :sparkling_heart:

**Comment by dennisdegreef** [comment]
Saturday deploy? :D

**Comment by WyriHaximus** [comment]
@dennisdegreef not really, now someone has to build a front onto this :wink: .

**Comment by rdohms** [comment]
well.. if the server goes belly up, we need support from True, since we are *at* true.. i guess we are safe :P

**Comment by dennisdegreef** [comment]
true

**Comment by rdohms** [comment]
yes, True

**Comment by dennisdegreef** [comment]
`$this->assertTrue($this);` ?

**Comment by rdohms** [comment]
image

**Comment by WyriHaximus** [comment]
Looks fishy

**Comment by dennisdegreef** [comment]
`!touch($this)`

**Comment by WyriHaximus** [comment]
Hammer!Time!

**Pull request merged**